src/main/java/net/geforcemods/securitycraft/items/TaserItem.java

```diff
@@ -73,9 +80,30 @@ else if(player.inventory.hasItemStack(oneRedstone))
 					return ActionResult.resultPass(stack);
 				}
 
-				WorldUtils.addScheduledTask(world, () -> world.addEntity(new TaserBulletEntity(world, player, powered)));
+				int range = 11;
+				Vector3d startVec = player.getEyePosition(1.0F);
+				Vector3d lookVec = player.getLook(1.0F).scale(range);
+				Vector3d endVec = startVec.add(lookVec);
+				AxisAlignedBB boundingBox = player.getBoundingBox().expand(lookVec).grow(1, 1, 1);
+				EntityRayTraceResult entityRayTraceResult = ProjectileHelper.rayTraceEntities(player, startVec, endVec, boundingBox, s -> s instanceof LivingEntity, range * range);
```

Comment:
This will crash on a server as `ProjectileHelper.rayTraceEntities` is client side only.

Reply:
Isn't there a `if (!world.isRemote)` check at the start of `onItemRightClick`?

Reply:
Yes, and `isRemote` is false on the server. This means the server tries to call `ProjectileHelper.rayTraceEntities`, which isn't there, and the game crashes.

Reply:
If you remove the check, the code will still be run on the server. There's no way around this other than copying the method into `TaserItem`.

More information on sides: https://jamieswhiteshirt.github.io/resources/the-client-server-architecture/

Reply:
And where does one check whether a method runs on the client/on the server/on both?

Reply:
A method runs on the client/server if it is called from there. The question is if a method is _physically present_ on the client/server. You can see that with the `@OnlyIn` annotation. Something that is annotated with `@OnlyIn(Side.CLIENT)` is only physically present on the client side, and something that is annotated with `@OnlyIn(Side.SERVER)` is only physically present on the server side.

Reply:
Yeah I haven't found a method in the `TaserItem.rayTraceEntities` method that has the `@OnlyIn(Side.CLIENT)` annotation, and I'll commit this now. If you find a method I've missed, please comment.